internal/backend/sessions/sessionworkflows/history.go

```diff
@@ -0,0 +1,385 @@
+		if types != 0 && types&sdktypes.CallAttemptStartSessionLogRecordType == 0 {
+			return sdktypes.InvalidSessionLogRecord, nil
+		}
+
+		attrs := a.ActivityTaskStartedEventAttributes
+		if attrs == nil {
+			l.Error("nil activity task started event attributes from temporal")
+			return sdktypes.InvalidSessionLogRecord, nil
+		}
+
+		if !wasRelevantEventCall(events, attrs.GetScheduledEventId()) {
+			return sdktypes.InvalidSessionLogRecord, nil
+		}
+
+		return sdktypes.NewCallAttemptStartSessionLogRecord(
+			t,
+			sdktypes.NewSessionCallAttemptStart(t, uint32(attrs.Attempt)),
+		), nil
+
+	case *historypb.HistoryEvent_ActivityTaskCompletedEventAttributes:
+		if types != 0 && types&sdktypes.CallAttemptCompleteSessionLogRecordType == 0 {
+			return sdktypes.InvalidSessionLogRecord, nil
+		}
+
+		attrs := a.ActivityTaskCompletedEventAttributes
+		if attrs == nil {
+			l.Error("nil activity task completed event attributes from temporal")
+			return sdktypes.InvalidSessionLogRecord, nil
+		}
+
+		if !wasRelevantEventCall(events, attrs.GetScheduledEventId()) {
+			return sdktypes.InvalidSessionLogRecord, nil
+		}
+
+		payloads := attrs.Result.GetPayloads()
+		if len(payloads) != 1 {
+			l.Error("unexpected number of payloads for activity task completed event", zap.Int("payloads", len(payloads)))
+			return sdktypes.InvalidSessionLogRecord, nil
+		}
+
+		var callOutputs sessioncalls.CallActivityOutputs
+
+		if err := json.Unmarshal(payloads[0].GetData(), &callOutputs); err != nil {
+			return sdktypes.InvalidSessionLogRecord, temporalclient.TranslateError(err, "unmarshal activity input")
+		}
+
+		return sdktypes.NewCallAttemptCompleteSessionLogRecord(
+			t,
+			sdktypes.NewSessionCallAttemptComplete(
+				t,
+				true,
+				callOutputs.Result,
+			),
+		), nil
+
+	case *historypb.HistoryEvent_ActivityTaskCanceledEventAttributes:
+		if types != 0 && types&sdktypes.CallAttemptCompleteSessionLogRecordType == 0 {
+			return sdktypes.InvalidSessionLogRecord, nil
+		}
+
+		attrs := a.ActivityTaskCanceledEventAttributes
+		if attrs == nil {
+			l.Error("nil activity task cancelled event attributes from temporal")
+			return sdktypes.InvalidSessionLogRecord, nil
+		}
+
+		if !wasRelevantEventCall(events, attrs.GetScheduledEventId()) {
+			return sdktypes.InvalidSessionLogRecord, nil
+		}
+
+		return sdktypes.NewCallAttemptCompleteSessionLogRecord(
+			t,
+			sdktypes.NewSessionCallAttemptComplete(
+				t,
+				true,
+				sdktypes.NewSessionCallAttemptResult(
+					sdktypes.InvalidValue,
+					workflow.ErrCanceled,
+				),
+			),
+		), nil
+
+	case *historypb.HistoryEvent_ActivityTaskFailedEventAttributes, *historypb.HistoryEvent_ActivityTaskTimedOutEventAttributes:
+		// do not report this, as it is always for infrastructure failures.
+		// we retry these indefinitely, so they are not interesting.
+		return sdktypes.InvalidSessionLogRecord, nil
+
+	default:
+		return sdktypes.InvalidSessionLogRecord, nil
```

Comment:
Log debugging data to warn us about the existence of this unexpected attribute type